<?xml version="1.0" encoding="UTF-8"?>
<testExecutions version="1">
  <file path="C:\Users\kevin\unirio\project-name\src\modules\cartao\cartao.repository.spec.ts">
    <testCase name="CartaoRepository insertCartao should insert Cartao" duration="6">
      <failure message="Error: expect(received).toEqual(expected) // deep equality"><![CDATA[Error: expect(received).toEqual(expected) // deep equality

Expected: ObjectContaining {"cvv": "123", "nomeTitular": "John Doe", "numero": "1234567890123456", "validade": "12/24"}
Received: undefined
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\cartao\cartao.repository.spec.ts:66:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)]]></failure>
    </testCase>
    <testCase name="CartaoRepository updateCartao should update Cartao by matricula" duration="1">
      <failure message="TypeError: Cannot read properties of undefined (reading &apos;id&apos;)"><![CDATA[TypeError: Cannot read properties of undefined (reading 'id')
    at CartaoRepository.id [as updateCartao] (C:\Users\kevin\unirio\project-name\src\modules\cartao\cartao.repository.ts:27:49)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\cartao\cartao.repository.spec.ts:89:23)]]></failure>
    </testCase>
    <testCase name="CartaoRepository updateCartao should return false when trying to update non-existent Cartao" duration="1">
      <failure message="TypeError: Cannot read properties of undefined (reading &apos;id&apos;)"><![CDATA[TypeError: Cannot read properties of undefined (reading 'id')
    at CartaoRepository.id [as updateCartao] (C:\Users\kevin\unirio\project-name\src\modules\cartao\cartao.repository.ts:27:49)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\cartao\cartao.repository.spec.ts:106:23)]]></failure>
    </testCase>
    <testCase name="CartaoRepository getCartaoByID should return Cartao by matricula" duration="2">
      <failure message="Error: expect(received).toEqual(expected) // deep equality"><![CDATA[Error: expect(received).toEqual(expected) // deep equality

Expected: {"cvv": "123", "id": 3, "nomeTitular": "John ew", "numero": "124213123123", "validade": "12/09"}
Received: undefined
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\cartao\cartao.repository.spec.ts:126:22)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)]]></failure>
    </testCase>
    <testCase name="CartaoRepository getCartaoByID should return undefined for non-existent Cartao" duration="1"/>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\devolucao\devolucao.repository.spec.ts">
    <testCase name="DevolucaoRepository should insert a new devolucao" duration="13">
      <failure message="Error: expect(received).toEqual(expected) // deep equality"><![CDATA[Error: expect(received).toEqual(expected) // deep equality

Expected: {"bicicleta": 9876, "ciclista": 1, "cobranca": 20.5, "horaFim": "2023-06-18T11:30:00", "horaInicio": "2023-06-18T10:00:00", "id": 1, "trancaFim": 5678, "trancaInicio": 1234}
Received: undefined
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\devolucao\devolucao.repository.spec.ts:58:20)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)]]></failure>
    </testCase>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\devolucao\devolucao.service.spec.ts">
    <testCase name="DevolucaoService insertDevolucao should insert a devolucao and return true" duration="217">
      <failure message="EntityMetadataNotFoundError: No metadata for &quot;Aluguel&quot; was found."><![CDATA[EntityMetadataNotFoundError: No metadata for "Aluguel" was found.
    at DataSource.getMetadata (C:\Users\kevin\unirio\project-name\src\data-source\DataSource.ts:444:30)
    at Repository.get metadata [as metadata] (C:\Users\kevin\unirio\project-name\src\repository\Repository.ts:53:40)
    at Repository.createQueryBuilder (C:\Users\kevin\unirio\project-name\src\repository\Repository.ts:82:18)
    at AluguelRepository.createQueryBuilder [as getAluguelByCiclista] (C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.repository.ts:24:49)
    at AluguelService.getAluguelByCiclista (C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.service.ts:63:46)
    at DevolucaoService.getAluguelByCiclista [as insertDevolucao] (C:\Users\kevin\unirio\project-name\src\modules\devolucao\devolucao.service.ts:22:44)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\devolucao\devolucao.service.spec.ts:103:45)
    at Promise.then.completed (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:293:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:226:10)
    at _callCircusTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:297:40)
    at _runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:233:3)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:135:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at run (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:68:3)
    at runAndTransformResultsToJestFormat (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapterInit.js:122:21)
    at jestAdapter (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapter.js:79:19)
    at runTestInternal (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:367:16)
    at runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:444:34)
    at Object.worker (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\testWorker.js:106:12)]]></failure>
    </testCase>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.service.spec.ts">
    <testCase name="AluguelService insertAluguel should insert a aluguel and return true" duration="230">
      <failure message="Error: expect(jest.fn()).toHaveBeenCalledWith(...expected)"><![CDATA[Error: expect(jest.fn()).toHaveBeenCalledWith(...expected)

- Expected
+ Received

- Object {
+ realizaCobranÃ§a {
    "ciclista": 1,
-   "valor": 30,
+   "valor": 10,
  },

Number of calls: 1
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.service.spec.ts:101:35)]]></failure>
    </testCase>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.service.spec.ts">
    <testCase name="CiclistaService insertCiclista should insert a ciclista and return true" duration="217">
      <failure message="TypeError: this.api.validaCartaoMock is not a function"><![CDATA[TypeError: this.api.validaCartaoMock is not a function
    at CiclistaService.validaCartaoMock [as insertCiclista] (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.service.ts:26:23)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.service.spec.ts:108:44)
    at Promise.then.completed (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:293:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:226:10)
    at _callCircusTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:297:40)
    at _runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:233:3)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:135:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at run (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:68:3)
    at runAndTransformResultsToJestFormat (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapterInit.js:122:21)
    at jestAdapter (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapter.js:79:19)
    at runTestInternal (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:367:16)
    at runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:444:34)
    at Object.worker (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\testWorker.js:106:12)]]></failure>
    </testCase>
    <testCase name="CiclistaService insertCiclista should throw NotFoundException if insertCiclista returns false" duration="43">
      <failure message="Error: expect(received).rejects.toThrow(expected)"><![CDATA[Error: expect(received).rejects.toThrow(expected)

Expected constructor: NotFoundException
Received constructor: TypeError

Received message: "this.api.validaCartaoMock is not a function"

      24 |   async insertCiclista(ciclista: CadastroCiclista): Promise<Ciclista> {
      25 |    
    > 26 |     if(await this.api.validaCartaoMock(ciclista.meioDePagamento)){
         |                       ^
      27 |    console.log("bateu")
      28 |     const check= await this.ciclistaRepository.insertCiclista(ciclista.ciclista)
      29 |     const card= await this.cartaoService.insertCartao(ciclista.meioDePagamento,check.id)

      at CiclistaService.validaCartaoMock [as insertCiclista] (src/modules/ciclista/ciclista.service.ts:26:23)
      at Object.<anonymous> (src/modules/ciclista/ciclista.service.spec.ts:132:36)
    at Object.toThrow (C:\Users\kevin\unirio\project-name\node_modules\expect\build\index.js:210:22)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.service.spec.ts:132:70)
    at Promise.then.completed (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:293:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:226:10)
    at _callCircusTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:297:40)
    at _runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:233:3)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:135:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at run (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:68:3)
    at runAndTransformResultsToJestFormat (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapterInit.js:122:21)
    at jestAdapter (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapter.js:79:19)
    at runTestInternal (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:367:16)
    at runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:444:34)
    at Object.worker (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\testWorker.js:106:12)]]></failure>
    </testCase>
    <testCase name="CiclistaService insertCiclista should throw NotFoundException if validaCartaoMock returns false" duration="13"/>
    <testCase name="CiclistaService updateCiclista should update a ciclista and return true" duration="11"/>
    <testCase name="CiclistaService updateCiclista should throw NotFoundException if updateCiclista returns false" duration="11"/>
    <testCase name="CiclistaService ativarCiclista should activate a ciclista and return true" duration="18"/>
    <testCase name="CiclistaService ativarCiclista should throw NotFoundException if ativarCiclista returns false" duration="11"/>
    <testCase name="CiclistaService deleteCiclista should delete a ciclista and return true" duration="19"/>
    <testCase name="CiclistaService deleteCiclista should throw NotFoundException if deleteCiclista returns false" duration="18"/>
    <testCase name="CiclistaService getCiclistaByID should get a ciclista by ID and return true" duration="30"/>
    <testCase name="CiclistaService getCiclistaByID should throw NotFoundException if getCiclistaByID returns null" duration="17"/>
    <testCase name="CiclistaService checkEmail should check email and return true" duration="13"/>
    <testCase name="CiclistaService checkEmail should throw BadRequestException if email is null or blank" duration="14"/>
    <testCase name="CiclistaService permiteAluguel should get a ciclista by ID and return true" duration="19"/>
    <testCase name="CiclistaService permiteAluguel should get a permiteAluguel by ID and return false" duration="20"/>
    <testCase name="CiclistaService permiteAluguel should throw NotFoundException if permiteAluguel returns null" duration="13"/>
    <testCase name="CiclistaService getBikeByCiclista should get a ciclista by ID and return true" duration="15">
      <failure message="TypeError: this.api.getBicicletaByid is not a function"><![CDATA[TypeError: this.api.getBicicletaByid is not a function
    at CiclistaService.getBicicletaByid [as getBikeByCiclista] (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.service.ts:123:35)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.service.spec.ts:324:22)]]></failure>
    </testCase>
    <testCase name="CiclistaService getBikeByCiclista should throw NotFoundException if getBikeByCiclista returns null" duration="10"/>
    <testCase name="CiclistaService getCiclistas should get all ciclistas" duration="12"/>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.spec.ts">
    <testCase name="CiclistaRepository should insert a new ciclista" duration="9">
      <failure message="Error: expect(received).toEqual(expected) // deep equality"><![CDATA[Error: expect(received).toEqual(expected) // deep equality

Expected: ObjectContaining {"cpf": "1234567890", "email": "john.doe@example.com", "nacionalidade": "BRASILEIRO", "nascimento": "1990-01-01", "nome": "John Doe", "passaporte": {"numero": "ABCD1234", "pais": "Brasil", "validade": "2025-12-31"}, "senha": "password123", "status": "AGUARDANDO_CONFIRMACAO", "urlFotoDocumento": "https://example.com/document.jpg"}
Received: undefined
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.spec.ts:77:20)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)]]></failure>
    </testCase>
    <testCase name="CiclistaRepository should update an existing ciclista" duration="1">
      <failure message="Error: Cannot spy on the findOneBy property because it is not a function; undefined given instead. If you are trying to mock a property, use `jest.replaceProperty(object, &apos;findOneBy&apos;, value)` instead."><![CDATA[Error: Cannot spy on the findOneBy property because it is not a function; undefined given instead. If you are trying to mock a property, use `jest.replaceProperty(object, 'findOneBy', value)` instead.
    at ModuleMocker.spyOn (C:\Users\kevin\unirio\project-name\node_modules\jest-mock\build\index.js:732:15)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.spec.ts:131:8)
    at Promise.then.completed (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:293:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:226:10)
    at _callCircusTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:297:40)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at _runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:233:3)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:135:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at run (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:68:3)
    at runAndTransformResultsToJestFormat (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapterInit.js:122:21)
    at jestAdapter (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapter.js:79:19)
    at runTestInternal (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:367:16)
    at runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:444:34)
    at Object.worker (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\testWorker.js:106:12)]]></failure>
    </testCase>
    <testCase name="CiclistaRepository should delete an existing ciclista" duration="1">
      <failure message="TypeError: this.ciclistaRepository.createQueryBuilder is not a function"><![CDATA[TypeError: this.ciclistaRepository.createQueryBuilder is not a function
    at CiclistaRepository.createQueryBuilder [as getCiclistaByID] (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.ts:49:6)
    at CiclistaRepository.getCiclistaByID [as deleteCiclista] (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.ts:32:38)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.spec.ts:159:45)
    at Promise.then.completed (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:293:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:226:10)
    at _callCircusTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:297:40)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at _runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:233:3)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:135:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at run (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:68:3)
    at runAndTransformResultsToJestFormat (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapterInit.js:122:21)
    at jestAdapter (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapter.js:79:19)
    at runTestInternal (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:367:16)
    at runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:444:34)
    at Object.worker (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\testWorker.js:106:12)]]></failure>
    </testCase>
    <testCase name="CiclistaRepository should get a ciclista by ID" duration="1">
      <failure message="TypeError: this.ciclistaRepository.createQueryBuilder is not a function"><![CDATA[TypeError: this.ciclistaRepository.createQueryBuilder is not a function
    at CiclistaRepository.createQueryBuilder [as getCiclistaByID] (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.ts:49:6)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.spec.ts:170:45)
    at Promise.then.completed (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:293:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:226:10)
    at _callCircusTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:297:40)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at _runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:233:3)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:135:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at run (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:68:3)
    at runAndTransformResultsToJestFormat (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapterInit.js:122:21)
    at jestAdapter (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapter.js:79:19)
    at runTestInternal (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:367:16)
    at runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:444:34)
    at Object.worker (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\testWorker.js:106:12)]]></failure>
    </testCase>
    <testCase name="CiclistaRepository should check if an email is available" duration="1">
      <failure message="TypeError: this.ciclistaRepository.findOneBy is not a function"><![CDATA[TypeError: this.ciclistaRepository.findOneBy is not a function
    at CiclistaRepository.findOneBy [as checkEmail] (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.ts:58:53)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.spec.ts:180:45)
    at Promise.then.completed (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:293:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:226:10)
    at _callCircusTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:297:40)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at _runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:233:3)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:135:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at run (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:68:3)
    at runAndTransformResultsToJestFormat (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapterInit.js:122:21)
    at jestAdapter (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapter.js:79:19)
    at runTestInternal (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:367:16)
    at runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:444:34)
    at Object.worker (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\testWorker.js:106:12)]]></failure>
    </testCase>
    <testCase name="CiclistaRepository should activate a ciclista" duration="1">
      <failure message="TypeError: this.ciclistaRepository.createQueryBuilder is not a function"><![CDATA[TypeError: this.ciclistaRepository.createQueryBuilder is not a function
    at CiclistaRepository.createQueryBuilder [as getCiclistaByID] (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.ts:49:6)
    at CiclistaRepository.getCiclistaByID [as ativarCiclista] (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.ts:67:39)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.repository.spec.ts:191:45)
    at Promise.then.completed (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:293:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:226:10)
    at _callCircusTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:297:40)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at _runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:233:3)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:135:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at run (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:68:3)
    at runAndTransformResultsToJestFormat (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapterInit.js:122:21)
    at jestAdapter (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapter.js:79:19)
    at runTestInternal (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:367:16)
    at runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:444:34)
    at Object.worker (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\testWorker.js:106:12)]]></failure>
    </testCase>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.repository.spec.ts">
    <testCase name="AluguelRepository should insert a new aluguel" duration="7">
      <failure message="Error: expect(received).toEqual(expected) // deep equality"><![CDATA[Error: expect(received).toEqual(expected) // deep equality

Expected: {"bicicleta": 9876, "ciclista": 1, "cobranca": 20.5, "horaFim": "2023-06-18T11:30:00", "horaInicio": "2023-06-18T10:00:00", "id": 1, "trancaFim": 5678, "trancaInicio": 1234}
Received: undefined
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.repository.spec.ts:59:20)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)]]></failure>
    </testCase>
    <testCase name="AluguelRepository should get a  Bike By Ciclista ID" duration="1">
      <failure message="TypeError: this.aluguelRepository.findOneBy is not a function"><![CDATA[TypeError: this.aluguelRepository.findOneBy is not a function
    at AluguelRepository.findOneBy [as getBikeByCiclista] (C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.repository.ts:41:49)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.repository.spec.ts:67:44)
    at Promise.then.completed (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:293:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:226:10)
    at _callCircusTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:297:40)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at _runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:233:3)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:135:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at run (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:68:3)
    at runAndTransformResultsToJestFormat (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapterInit.js:122:21)
    at jestAdapter (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapter.js:79:19)
    at runTestInternal (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:367:16)
    at runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:444:34)
    at Object.worker (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\testWorker.js:106:12)]]></failure>
    </testCase>
    <testCase name="AluguelRepository should check if permiteAluguel already exist aluguel" duration="1">
      <failure message="TypeError: this.aluguelRepository.findOneBy is not a function"><![CDATA[TypeError: this.aluguelRepository.findOneBy is not a function
    at AluguelRepository.findOneBy [as permiteAluguel] (C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.repository.ts:32:49)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.repository.spec.ts:78:44)
    at Promise.then.completed (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:293:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:226:10)
    at _callCircusTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:297:40)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at _runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:233:3)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:135:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at run (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:68:3)
    at runAndTransformResultsToJestFormat (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapterInit.js:122:21)
    at jestAdapter (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapter.js:79:19)
    at runTestInternal (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:367:16)
    at runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:444:34)
    at Object.worker (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\testWorker.js:106:12)]]></failure>
    </testCase>
    <testCase name="AluguelRepository should check if permiteAluguel not already exist aluguel " duration="1">
      <failure message="TypeError: this.aluguelRepository.findOneBy is not a function"><![CDATA[TypeError: this.aluguelRepository.findOneBy is not a function
    at AluguelRepository.findOneBy [as permiteAluguel] (C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.repository.ts:32:49)
    at Object.<anonymous> (C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.repository.spec.ts:90:44)
    at Promise.then.completed (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:293:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\utils.js:226:10)
    at _callCircusTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:297:40)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at _runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:233:3)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:135:9)
    at _runTestsForDescribeBlock (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:130:9)
    at run (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\run.js:68:3)
    at runAndTransformResultsToJestFormat (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapterInit.js:122:21)
    at jestAdapter (C:\Users\kevin\unirio\project-name\node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapter.js:79:19)
    at runTestInternal (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:367:16)
    at runTest (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\runTest.js:444:34)
    at Object.worker (C:\Users\kevin\unirio\project-name\node_modules\jest-runner\build\testWorker.js:106:12)]]></failure>
    </testCase>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\devolucao\devolucao.controller.spec.ts">
    <testCase name="DevolucaoController cadastrarDevolucao should call insertDevolucao in the service and return response from service" duration="142"/>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\ciclista\ciclista.controller.spec.ts">
    <testCase name="CiclistaController cadastrarCiclista should call insertCiclista in the service and return response from service" duration="128"/>
    <testCase name="CiclistaController getCiclista should call getCiclistas in the service and return response from service" duration="17"/>
    <testCase name="CiclistaController updateCiclista should call updateCiclista in the service and return response from service" duration="16"/>
    <testCase name="CiclistaController deleteCiclista should call deleteCiclista in the service and return response from service" duration="13"/>
    <testCase name="CiclistaController getCiclistaByID should call getCiclistaByID in the service and return response from service" duration="15"/>
    <testCase name="CiclistaController ativarCiclista should call ativarCiclista in the service and return response from service" duration="11"/>
    <testCase name="CiclistaController checkEmail should call checkEmail in the service and return response from service" duration="17"/>
    <testCase name="CiclistaController permiteAluguel should call permiteAluguel in the service and return response from service" duration="10"/>
    <testCase name="CiclistaController getBikeByCiclista should call getBikeByCiclista in the service and return response from service" duration="21"/>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\funcionario\funcionario.controller.spec.ts">
    <testCase name="FuncionarioController cadastrarFuncionario should call insertFuncionario in the service and return response from service" duration="121"/>
    <testCase name="FuncionarioController getFuncionario should call getFuncionarios in the service and return response from service" duration="13"/>
    <testCase name="FuncionarioController updateFuncionario should call updateFuncionario in the service and return response from service" duration="12"/>
    <testCase name="FuncionarioController deleteFuncionario should call deleteFuncionario in the service and return response from service" duration="12"/>
    <testCase name="FuncionarioController getFuncionarioByID should call getFuncionarioByID in the service and return response from service" duration="10"/>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\aluguel\aluguel.controller.spec.ts">
    <testCase name="AluguelController cadastrarAluguel should call insertAluguel in the service and return response from service" duration="143"/>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\funcionario\funcionario.service.spec.ts">
    <testCase name="FuncionarioService insertFuncionario should insert Funcionario" duration="7"/>
    <testCase name="FuncionarioService updateFuncionario should update Funcionario" duration="1"/>
    <testCase name="FuncionarioService updateFuncionario should throw NotFoundException when Funcionario is not found" duration="19"/>
    <testCase name="FuncionarioService updateFuncionario should not update Funcionario when novoFuncionario is null or blank" duration="1"/>
    <testCase name="FuncionarioService deleteFuncionario should delete Funcionario" duration="1"/>
    <testCase name="FuncionarioService deleteFuncionario should throw NotFoundException when Funcionario is not found" duration="1"/>
    <testCase name="FuncionarioService deleteFuncionario should not delete Funcionario when id is null or blank" duration="1"/>
    <testCase name="FuncionarioService getFuncionarioByID should return Funcionario by id" duration="1"/>
    <testCase name="FuncionarioService getFuncionarioByID should throw NotFoundException when Funcionario is not found" duration="1"/>
    <testCase name="FuncionarioService getFuncionarioByID should not getFuncionarioByID when id is null or blank" duration="1"/>
    <testCase name="FuncionarioService getFuncionarios should return all Funcionarios" duration="2"/>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\cartao\cartao.controller.spec.ts">
    <testCase name="CartaoController updateCartao should call updateCartao in the service and return response from service" duration="154"/>
    <testCase name="CartaoController getCartaoByID should call getCartaoByID in the service and return response from service" duration="19"/>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\cartao\cartao.service.spec.ts">
    <testCase name="CartaoService insertCartao should insert Cartao" duration="4"/>
    <testCase name="CartaoService updateCartao should update Cartao" duration="2"/>
    <testCase name="CartaoService updateCartao should throw NotFoundException when Cartao is not found" duration="7"/>
    <testCase name="CartaoService getCartaoByID should return Cartao by id" duration="1"/>
    <testCase name="CartaoService getCartaoByID should throw NotFoundException when Cartao is not found" duration="6"/>
    <testCase name="CartaoService getCartaoByID should not getCartaoByID when id is null or blank" duration="1"/>
  </file>
  <file path="C:\Users\kevin\unirio\project-name\src\modules\funcionario\funcionario.repository.spec.ts">
  </file>
</testExecutions>